File: evals/validation/e1/run-2026-07-26/divergence/prepare-blind-inputs.mjs
#!/usr/bin/env node
// prepare-blind-inputs.mjs — builds the per-judge blinded input sets for the
// E2 blind protocols (e: intent matching; b: layout class; c: visual motifs)
// and for the E1 rubric jurors. Deterministic seeded shuffles; seeds recorded
// in the output JSON (registered requirement: "seeds recorded").
//
// Shuffle: Fisher-Yates driven by mulberry32(seed). Seeds fixed here, in the
// committed script, before any judging: judge k (1..5) uses seed 202607260+k
// for the RENDER order and seed 202607360+k for the INTENT order; rubric juror
// k uses seed 202607460+k for arm presentation order.
//
// Usage: node prepare-blind-inputs.mjs <rendersDir> <outDir>
//   rendersDir: contains <arm>__<vp>__<mode>.png cells
//   outDir: writes judge-k/ and juror-k/ dirs with neutral-named copies +
//           blind-map.json (the seed record; committed AFTER judging so the
//           mapping is auditable, never shown to judges).
import { readdirSync, mkdirSync, copyFileSync, writeFileSync } from "node:fs";
import { join } from "node:path";

const ARMS = ["trail-ledger", "status-board", "first-light"]; // fixed order: alphabetical of dir creation order in DIRECTIONS.md
function mulberry32(a) {
  return function () {
    a |= 0; a = (a + 0x6D2B79F5) | 0;
    let t = Math.imul(a ^ (a >>> 15), 1 | a);
    t = (t + Math.imul(t ^ (t >>> 7), 61 | t)) ^ t;
    return ((t ^ (t >>> 14)) >>> 0) / 4294967296;
  };
}
function shuffled(arr, seed) {
  const rng = mulberry32(seed), out = [...arr];
  for (let i = out.length - 1; i > 0; i--) {
    const j = Math.floor(rng() * (i + 1));
    [out[i], out[j]] = [out[j], out[i]];
  }
  return out;
}

const [rendersDir, outDir] = process.argv.slice(2);
if (!rendersDir || !outDir) { console.error("usage: node prepare-blind-inputs.mjs <rendersDir> <outDir>"); process.exit(2); }
const cells = readdirSync(rendersDir).filter((f) => f.endsWith(".png"));
const record = { armOrderBase: ARMS, judges: {}, jurors: {} };

for (let k = 1; k <= 5; k++) {
  // E2 blind judges: desktop-light full-page renders only, neutral names R1..R3
  const renderSeed = 202607260 + k, intentSeed = 202607360 + k;
  const renderOrder = shuffled(ARMS, renderSeed);
  const intentOrder = shuffled(ARMS, intentSeed);
  const dir = join(outDir, `judge-${k}`);
  mkdirSync(dir, { recursive: true });
  renderOrder.forEach((arm, i) => {
    const src = cells.find((c) => c.startsWith(`${arm}__desktop__light`));
    if (!src) throw new Error(`missing desktop-light cell for ${arm}`);
    copyFileSync(join(rendersDir, src), join(dir, `render-${i + 1}.png`));
  });
  record.judges[`judge-${k}`] = { renderSeed, intentSeed, renderOrder, intentOrder };

  // E1 rubric jurors: full matrix per arm, neutral names candidate-1..3
  const jurorSeed = 202607460 + k;
  const armOrder = shuffled(ARMS, jurorSeed);
  const jdir = join(outDir, `juror-${k}`);
  mkdirSync(jdir, { recursive: true });
  armOrder.forEach((arm, i) => {
    for (const cell of cells.filter((c) => c.startsWith(`${arm}__`))) {
      copyFileSync(join(rendersDir, cell), join(jdir, cell.replace(`${arm}__`, `candidate-${i + 1}__`)));
    }
  });
  record.jurors[`juror-${k}`] = { jurorSeed, armOrder };
}
writeFileSync(join(outDir, "blind-map.json"), JSON.stringify(record, null, 2) + "\n");
console.log("blind inputs prepared:", JSON.stringify(Object.keys(record.judges)));

File: evals/validation/e3/run-2026-07-26-editorial/jurors/prepare-blind-inputs.py
#!/usr/bin/env python3
"""E3 editorial panel prep — deterministic blind inputs.

Writes jurors/blind-map.json (seeded per-juror arm shuffles), one
juror-K-inputs.md per juror (candidate briefs under neutral labels, in that
juror's order: frozen intent + A1 declared breaks + machine-gate outcome), and
copies the render matrix into a per-juror scratch directory under neutral
candidate names (arm filenames would unblind the panel; the scratch copies are
byte-copies of the committed renders and are not committed themselves).

Usage: python prepare-blind-inputs.py <scratch-blind-dir>
"""
import json, random, shutil, sys
from pathlib import Path

HERE = Path(__file__).resolve().parent
PROJECT = HERE.parent / "project"
ARMS = ["almanac-rail", "weir-plates", "two-inks"]
SEED_BASE = 202607560  # E3-editorial juror seed family (E1 used 202607460+)

INTENTS = {
    "almanac-rail": "A parish almanac read top to bottom: every year since 1990 is one ruled line whose rain marks sit on a fixed September-to-October shelf, so the eye watches the wet season walk later down the page while the entries grow heavier.",
    "weir-plates": "Three patient broadsheet plates — when the rain comes, how much falls, how hard the worst day hits — each annotated in the reporter's own words, with the prose walking the reader from plate to plate.",
    "two-inks": "Two inks for two instruments: the manual-gauge years print in umber, the weir years in river blue, and the whole story — later, harder — is read by watching where the blue marks drift away from the umber ones.",
}
BREAKS = {
    "almanac-rail": [
        "The year-list IS the only chart — no separate hero graphic; the almanac rail (34 ruled year-rows, onset mark on a shared calendar shelf, season-total bar, heaviest-day figure) carries the entire quantitative story (in an almanac the record itself is the reading matter; splitting chart from table would make the data decoration).",
        "Editorial interjections interrupt the record — the rail breaks at the decade turns, the gaps, the method change, and 2019 for short prose asides, so caveats and story live inside the record, not in footnotes (the local paper's patience made visible).",
    ],
    "weir-plates": [
        "The journalism is written on the plates — annotations (the method change, the gaps, the 2019 flag, the era means) are keyed directly to each chart plate instead of a separate caption column (if every device must trace to the reporting, the strongest form is reporting inked onto the graphic itself).",
        "The grid is suppressed except where the story points — each plate draws only the reference lines its annotations name (era means, calendar shelf, gap markers), not a full axis lattice (a patient local paper prints nothing it does not need).",
    ],
    "two-inks": [
        "The method-change disclosure is promoted to the page's color system — every data mark, figure, and era label is inked by instrument (umber = manual gauge 1990-1996, blue = automated weir 1997-2023), never by color alone (era names written out at every crossing) — the data-integrity caveat becomes the brand layer.",
        "The finding leads; the record follows — the page opens with a facing then/now diptych (early manual-gauge era beside the recent verified weir years) before any time series, breaking the chronological-first convention (the reader's question is answered first; the 34-year evidence is laid out underneath for checking).",
    ],
}
GATE_LINE = (
    "All shipped machine gates GREEN for this candidate: token-contract AA recompute "
    "(text>=4.5, graphic>=3.0, light+dark); structural output floor (single main, h1, heading "
    "order, meta description); axe-core 0 serious / 0 critical on all 4 rendered cells; no "
    "horizontal overflow at 280/320/414px; state-aware contrast (default/hover/focus, both "
    "modes) pass; target-size (WCAG 2.5.8) pass; focus-trap not-applicable (no dialog); "
    "derived-claims honesty gate pass (34 year blocks scoped, both modes); required-content "
    "manifest pass (17 items, both modes). Mode fidelity verified ok for light and dark renders."
)
CELLS = ["desktop__light", "desktop__dark", "mobile__light", "mobile__dark"]


def main(blind_root: Path) -> None:
    jurors = {}
    for k in range(1, 6):
        rng = random.Random(SEED_BASE + k)
        order = ARMS[:]
        rng.shuffle(order)
        jurors[f"juror-{k}"] = {"jurorSeed": SEED_BASE + k, "armOrder": order}
    (HERE / "blind-map.json").write_text(
        json.dumps({"armOrderBase": ARMS, "jurors": jurors}, indent=2) + "\n", encoding="utf-8")

    for k in range(1, 6):
        order = jurors[f"juror-{k}"]["armOrder"]
        lines = ["# Candidate briefs (neutral labels; order fixed for you)\n"]
        jdir = blind_root / f"juror-{k}"
        jdir.mkdir(parents=True, exist_ok=True)
        for i, arm in enumerate(order, 1):
            lines.append(f"\n## candidate-{i}\n")
            lines.append(f'\nDirection intent (written before any code existed): "{INTENTS[arm]}"\n')
            lines.append(
                "\nDeclared convention breaks (per the pre-registration's Amendment A1, these and ONLY "
                "these are the candidate's deliberate breaks; any other convention break you find is "
                "undeclared and scores as a mistake, not a signature):")
            for b in BREAKS[arm]:
                lines.append(f"\n- {b}")
            lines.append(f"\n\nMachine-gate outcome: {GATE_LINE}\n")
            for cell in CELLS:
                shutil.copyfile(PROJECT / "renders" / f"{arm}__{cell}.png",
                                jdir / f"candidate-{i}__{cell}.png")
        (HERE / f"juror-{k}-inputs.md").write_text("".join(lines), encoding="utf-8")
    print("blind-map.json + 5 inputs files written; blind renders under", blind_root)


if __name__ == "__main__":
    main(Path(sys.argv[1]))
